modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc

```diff
@@ -393,14 +393,8 @@ class NDB_Menu_Filter_Form_Resolved_Violations extends NDB_Menu_Filter_Form
             }
         } else {
             // allow only to view own site data
-            $sites    = array();
-            $site_arr = $user->getData('CenterIDs');
-            foreach ($site_arr as $key => $val) {
-                $site[$key] = &Site::singleton($val);
-                if ($site[$key]->isStudySite()) {
-                    $sites[$val] = $site[$key]->getCenterName();
-                }
-            }
+            $sites = array();
```

Comment:
What's the point of this line if it gets overridden immediately below? (Same with above)

Reply:
@driusan 

the reason being that the function returns the sites of a user that are study sites. If the user is in one site that is a non study site, don't I need to initialize the array to the empty array so we avoid a 500 error?

Reply:
never mind my comment, I just checked that the function in the User class initialize the array so I assume it will return the empty array without me having to initialize it here, right?

Reply:
Even if the function returned something different, the return from the User class would overwrite the empty array with whatever was returned, since it's reassigned the line immediately below..

Reply:
done!